src/overlays/actors/ovl_En_Goroiwa/z_en_goroiwa.c

```diff
@@ -21,77 +42,695 @@ const ActorInit En_Goroiwa_InitVars = {
+
+    if (this->nextWaypoint < 0) {
+        if (temp_v0 == 0 || temp_v0 == 1) {
+            this->currentWaypoint = this->endWaypoint;
+            this->nextWaypoint = this->endWaypoint - 1;
+            this->pathDirection = -1;
+        } else if (temp_v0 == 3) {
+            this->currentWaypoint = 0;
+            this->nextWaypoint = 1;
+            this->pathDirection = 1;
+        }
+    } else if (this->nextWaypoint > this->endWaypoint) {
+        if (temp_v0 == 0 || temp_v0 == 1) {
+            this->currentWaypoint = 0;
+            this->nextWaypoint = 1;
+            this->pathDirection = 1;
+        } else if (temp_v0 == 3) {
+            this->currentWaypoint = this->endWaypoint;
+            this->nextWaypoint = this->endWaypoint - 1;
+            this->pathDirection = -1;
+        }
+    }
+}
+
+void func_80A4C134(EnGoroiwa* this) {
+    this->currentWaypoint = this->nextWaypoint;
+    this->nextWaypoint += this->pathDirection;
+    func_80A4C080(this);
+}
+
+void func_80A4C164(EnGoroiwa* this) {
+    this->pathDirection *= -1;
+    this->currentWaypoint = this->nextWaypoint;
+    this->nextWaypoint += this->pathDirection;
+}
+
+void func_80A4C188(EnGoroiwa* this, GlobalContext* globalCtx) {
+    this->endWaypoint = globalCtx->setupPathList[this->actor.params & 0xFF].count - 1;
+    this->currentWaypoint = 0;
+    this->nextWaypoint = 1;
+    this->pathDirection = 1;
+}
+
+void func_80A4C1C4(EnGoroiwa* this, GlobalContext* globalCtx, s32 waypoint) {
+    Path* path = &globalCtx->setupPathList[this->actor.params & 0xFF];
+    Vec3s* pointPos = (Vec3s*)SEGMENTED_TO_VIRTUAL(path->points) + waypoint;
+
+    this->actor.posRot.pos.x = pointPos->x;
+    this->actor.posRot.pos.y = pointPos->y;
+    this->actor.posRot.pos.z = pointPos->z;
+}
+
+void func_80A4C264(EnGoroiwa* this) {
+    this->unk_1B0.x = 1.0f;
+    this->unk_1C0 = 1.0f;
+}
+
+s32 func_80A4C27C(EnGoroiwa* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Path* path = &globalCtx->setupPathList[this->actor.params & 0xFF];
+    Vec3s* nextPointPos = (Vec3s*)SEGMENTED_TO_VIRTUAL(path->points) + this->nextWaypoint;
+    Vec3s* currentPointPos = (Vec3s*)SEGMENTED_TO_VIRTUAL(path->points) + this->currentWaypoint;
+
+    if (nextPointPos->x == currentPointPos->x && nextPointPos->z == currentPointPos->z) {
+        if (nextPointPos->y == currentPointPos->y) {
+            // Translation: Error: Invalid path data (points overlap)
+            osSyncPrintf("Error : レールデータ不正(点が重なっている)");
+            osSyncPrintf("(%s %d)(arg_data 0x%04x)\n", "../z_en_gr.c", 559, this->actor.params);
+        }
+
+        if (nextPointPos->y > currentPointPos->y) {
+            return 1;
+        } else {
+            return -1;
+        }
+    }
+
+    return 0;
+}
+
+void func_80A4C3A4(GlobalContext* globalCtx, Vec3f* arg1) {
+    static Vec3f effectVelocity = { 0.0f, 0.0f, 0.0f };
+    static Vec3f effectAccel = { 0.0f, 0.3f, 0.0f };
+
+    Vec3f effectPos;
+    s32 i;
```

Comment:
gonna still merge, but just as a general note we dont seperate out static and non static variables in the list. sometimes they dont always fit at the top of the list and it doesnt make sense to treat them separately. if you do end up fixing this before i get to a merge, its relevant in a few other places in the file :D